### PR TITLE
Feature/operator functions

### PR DESCRIPTION
**Pull request by docteurklein**

### CODE REVIEW

**Comment by docteurklein** [comment]
I must admit the copyright headers kills readability

File: src/Functional/Neq.php [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+ * @param mixed $a
+ * @param mixed $b
+ * @return boolean
+ */
+function neq($a, $b)
+{
+    return $a != $b;
+
+    // OR
+    //return not('Functional\eq')($a, $b);
```
Comment (docteurklein) [inline src/Functional/Neq.php]: @lstrojny what should I do here?

Comment (lstrojny) [inline src/Functional/Neq.php]: I would go with the best performing implementation, therefore `!==` (we should go with strict equality everywhere). Also: shouldn't this function also return a function, not the actual result?

Comment (docteurklein) [inline src/Functional/Neq.php]: ok with the best performing impl (`!=`).

Concerning strict equality (`!==`), I think we should provide both (which I did via the `is` method).

Concerning returning a function, I was wondering the same, but then when will we actually invoke the final function to get the result?

Comment (lstrojny) [inline src/Functional/Neq.php]: Identity vs equality: makes sense!

Returning a function: most likely we would wanna use it with e.g. `map()`, so this one would invoke the function.

**Comment by lstrojny** [comment]
Can you rebase this change?

**Comment by docteurklein** [comment]
@lstrojny I just resolved conflicts. It appears someone implemented `not` in the meanwhile.

**Comment by lstrojny** [comment]
@docteurklein going through the changes again I realize that we have `identical` now as well which duplicates `is`. If you are still interested, an "equal" function is still missing though. Check out `identical` for inspiration.

Sorry for the confusion.

**Comment by docteurklein** [comment]
no problem :) I'll try to de-dup this soon!

**Comment by lstrojny** [comment]
Closing for inactivity

**Pull request closed without merging**